kpet/utils.py

```diff
@@ -33,3 +37,106 @@ def get_template_content(filename, dbdir=None):
     template_path = get_template_path(filename, dbdir)
     with open(template_path) as file_handler:
         return file_handler.read()
+
+
+def print2(s):
+    sys.stderr.write(s + '\n')
+
+
+def smart_layout_insert_prefix(f_layout, obj):
+    if obj is None:
+        return None
+    prefix = os.path.dirname(os.path.realpath(f_layout))
+    return '%s/%s' % (prefix, str(obj))
+
+
+def smart_get_layout(f_layout):
+    text = ''
+    with open(f_layout, 'r') as f:
+        text = ''.join(f.readlines())
+
+    layout = json.loads(text, object_pairs_hook=collections.OrderedDict)
+    return layout
+
+
+def smart_get_patterns_bylayout(d_layout, f_layout):
+    l_patterns = []
+    for key in d_layout:
+        val = d_layout[key]
+        if val is None:
+            continue
+
+        s_file = smart_layout_insert_prefix(f_layout, val)
+        text = ''
+        with open(s_file, 'r') as f:
+            text = ''.join(f.readlines())
+
+        p = json.loads(text, object_pairs_hook=collections.OrderedDict)
+        for e in p['patterns']:
```

Comment:
Please don't use short names for variables. Code is written once and read thousands, use the best name that come to your mind, you will never regret spending time on this :).